src/keystore.c

```diff
@@ -208,22 +208,35 @@ static keystore_error_t _get_and_decrypt_seed(
 }
 
 static bool _verify_seed(
-    const char* password,
+    const uint8_t* encryption_key,
     const uint8_t* expected_seed,
     size_t expected_seed_len)
 {
-    uint8_t decrypted_seed[KEYSTORE_MAX_SEED_LENGTH] = {0};
-    size_t seed_len;
-    UTIL_CLEANUP_32(decrypted_seed);
-    if (_get_and_decrypt_seed(password, decrypted_seed, &seed_len, NULL) != KEYSTORE_OK) {
+    uint8_t encrypted_seed_and_hmac[96];
+    UTIL_CLEANUP_32(encrypted_seed_and_hmac);
+    uint8_t encrypted_len;
+    if (!memory_get_encrypted_seed_and_hmac(encrypted_seed_and_hmac, &encrypted_len)) {
         return false;
     }
-    if (expected_seed_len != seed_len) {
+    if (encrypted_len < 49) {
+        Abort("_verify_seed: underflow / zero size");
+    }
+    size_t decrypted_len = encrypted_len - 48;
+    uint8_t decrypted[decrypted_len];
```

Comment:
it is better to have a fixed upper bound instead of dynamically allocating stack size.

Reply:
It's a copypaste of https://github.com/BitBoxSwiss/bitbox02-firmware/blob/dea7eff17b278e399822a2b2a2aa3ef45f48e11c/src/keystore.c#L191. I prefer to not change it in this PR, or not at all and port it to Rust (https://github.com/BitBoxSwiss/bitbox02-firmware/pull/1425)

Reply:
`decrypted_len` is a `size_t` AFAICT